src/Microsoft.ML.Core/Data/DataKind.cs

```diff
@@ -8,40 +8,83 @@
 namespace Microsoft.ML.Data
 {
     /// <summary>
-    /// Data type specifier.
+    /// Data type specifier used in text loader and type converters.
```

Comment:
I don't really like giving where a type is used as part of its definition/summary of what it is. What happens if a new place starts using this enum?

Reply:
Then we need to update this doc string. We have this new enum only because of the two classes mentioned.

---
In reply to: [259519002](https://github.com/dotnet/machinelearning/pull/2661#discussion_r259519002) [](ancestors = 259519002)

Reply:
> We have this new enum only because of the two classes mentioned.

Right now, sure. But nothing is stopping some new API from using this enum. Then this doc would be out of date. Saying where something is used now doesn't explain what this thing is.

How about simply:

```
Specifies a simple data type.
```

other enum summaries are pretty simple:

https://docs.microsoft.com/en-us/dotnet/api/system?view=netframework-4.7.2#enums

>  TypeCode | Specifies the type of an object.
>  DayOfWeek | Specifies the day of the week.

If you want to add where it is used, that would be better in a `<remarks>` section.

Reply:
Ok. I will move it to `//`.

---
In reply to: [259536248](https://github.com/dotnet/machinelearning/pull/2661#discussion_r259536248) [](ancestors = 259536248)